File: java-app/src/main/java/com/dexter/utils/UploadReceiver.java
package com.dexter.utils;

import android.content.Context;

import net.gotev.uploadservice.ServerResponse;
import net.gotev.uploadservice.UploadInfo;
import net.gotev.uploadservice.UploadServiceBroadcastReceiver;

/**
 * A class to track a specific upload
 */
public class UploadReceiver extends UploadServiceBroadcastReceiver {

    /**
     * A method which indicates the progress of the upload
     *
     * @param context    the context of the class
     * @param uploadInfo the information on the upload
     */
    @Override
    public void onProgress(Context context, UploadInfo uploadInfo) {
        super.onProgress(context, uploadInfo);
    }

    /**
     * A method which indicates that there is some error occurred while upload
     *
     * @param context        the context of the class
     * @param uploadInfo     the information on the upload
     * @param serverResponse the response returned by server
     * @param exception      the exception occurred
     */
    @Override
    public void onError(Context context, UploadInfo uploadInfo, ServerResponse serverResponse, Exception exception) {
        super.onError(context, uploadInfo, serverResponse, exception);
    }

    /**
     * A method which indicates that the upload is successful
     *
     * @param context        the context of the class
     * @param uploadInfo     the information on the upload
     * @param serverResponse the response returned by server
     */
    @Override
    public void onCompleted(Context context, UploadInfo uploadInfo, ServerResponse serverResponse) {
        super.onCompleted(context, uploadInfo, serverResponse);
    }

    /**
     * A method which indicates that upload is cancelled
     *
     * @param context    the context of the class
     * @param uploadInfo the information on the upload
     */
    @Override
    public void onCancelled(Context context, UploadInfo uploadInfo) {
        super.onCancelled(context, uploadInfo);
    }
}

File: kotlin-app/src/main/java/com/dexter/utils/UploadReceiver.kt
package com.dexter.utils

import android.content.Context

import net.gotev.uploadservice.ServerResponse
import net.gotev.uploadservice.UploadInfo
import net.gotev.uploadservice.UploadServiceBroadcastReceiver

/**
 * A class to track a specific upload
 *
 *
 * use register() function to register the receiver
 * use unregister() function to unregister the receiver, if already registered
 */
class UploadReceiver : UploadServiceBroadcastReceiver() {

    /**
     * A function which indicates the progress of the upload
     *
     * @param context    the context of the class
     * @param uploadInfo the information on the upload
     */
    override fun onProgress(context: Context?, uploadInfo: UploadInfo?) {
        super.onProgress(context, uploadInfo)
    }

    /**
     * A function which indicates that there is some error occurred while upload
     *
     * @param context        the context of the class
     * @param uploadInfo     the information on the upload
     * @param serverResponse the response returned by server
     * @param exception      the exception occurred
     */
    override fun onError(context: Context?, uploadInfo: UploadInfo?, serverResponse: ServerResponse?, exception: Exception?) {
        super.onError(context, uploadInfo, serverResponse, exception)
    }

    /**
     * A function which indicates that the upload is successful
     *
     * @param context        the context of the class
     * @param uploadInfo     the information on the upload
     * @param serverResponse the response returned by server
     */
    override fun onCompleted(context: Context?, uploadInfo: UploadInfo?, serverResponse: ServerResponse?) {
        super.onCompleted(context, uploadInfo, serverResponse)
    }

    /**
     * A function which indicates that upload is cancelled
     *
     * @param context    the context of the class
     * @param uploadInfo the information on the upload
     */
    override fun onCancelled(context: Context?, uploadInfo: UploadInfo?) {
        super.onCancelled(context, uploadInfo)
    }
}